test/dns_dnscmd/dnscmd_test.rb

```diff
@@ -13,51 +15,51 @@ def test_dnscmd_provider_initialization
   end
 
   def test_create_address_record
-    Proxy::Dns::Dnscmd::Record.any_instance.expects(:dns_find).with('host.domain').returns(false)
-    Proxy::Dns::Dnscmd::Record.any_instance.expects(:execute).with('/RecordAdd domain host.domain. A 192.168.33.33', anything).returns(true)
-    assert Proxy::Dns::Dnscmd::Record.new.create_a_record('host.domain', '192.168.33.33')
+    Proxy::Dns::Dnscmd::Record.any_instance.expects(:dns_find).with('host.foo.bar.domain.local').returns(false)
+    Proxy::Dns::Dnscmd::Record.any_instance.expects(:execute).with('/RecordAdd bar.domain.local host.foo.bar.domain.local. A 192.168.33.33', anything).returns(true)
```

Comment:
Indeed, dnscmd just adds the zone if it does not exist; hence `host.foo` is a perfectly valid hostname. Since we are are explicitly defining a FQDN with  `host.foo.bar.domain.local.` our A record would be created with the expected result (*). 
Sine users are going to create their sub domains and zones in foreman, I think we can safely leverage the resulting domain to users: We should get  `host.foo` if we have actually created a zone named `foo` in foreman, resulting in the desired behavior?

(*) Using a relative domain name with dnscmd would create the host in the defined domain; eg 
`/RecordAdd bar.domain.local host.foo.bar.domain.local A 192.168.33.33` would result in the FQDN: `host.foo.bar.domain.local.bar.domain.local`